stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -251,6 +315,48 @@ inline bool AsyncTask::isCancelled() const {
                           .isCancelled();
 }
 
+inline void AsyncTask::flagAsRunning() {
+  auto oldStatus = _private().Status.load(std::memory_order_relaxed);
+  while (true) {
+    assert(!oldStatus.isRunning());
+    if (oldStatus.isLocked()) {
+      return flagAsRunning_slow();
+    }
+
+    auto newStatus = oldStatus.withRunning(true);
+    if (newStatus.isStoredPriorityEscalated()) {
+      newStatus = newStatus.withoutStoredPriorityEscalation();
+      Flags.setPriority(oldStatus.getStoredPriority());
+    }
+
+    if (_private().Status.compare_exchange_weak(oldStatus, newStatus,
+                                                std::memory_order_relaxed,
+                                                std::memory_order_relaxed))
+      return;
+  }
```

Comment:
Nevermind previous comment here; we do load again relaxed, that's the compare_exchange's api... 😉